src/data/target.ts

```diff
@@ -10,15 +10,20 @@ export function reset(): void {
     queue = {};
 }
 
-export function observe(id: number): number {
-    if (id !== null && !(id in queue)) {
-        let value = dom.getValue(id);
-        let node = dom.getNode(id) as Element;
-        queue[id] = {
-            id,
-            hash: value ? hash(value.selector) : "",
-            box: node && node.nodeType !== Node.TEXT_NODE ? layout(node) : null
-        };
+export function observe(node: Node): number {
+    let id = null;
+    if (node !== null) {
+        let value = dom.get(node);
+        if (value !== null) {
+            id = value.id;
+            if (id !== null && !(id in queue)) {
+                queue[id] = {
```

Comment:
How does the value get "fixed up" in this case (i.e. when value.id === null? Or once it gets into the queue?

Reply:
The assumption here is that we will get to "observe" only after all previous DOM operations have completed (discover / mutations). So when we come here, the value should be present - of course we will know more from the real data that comes back. We will receive null value only for unexpected cases at this time, e.g. elements that we are not tracking for some reason but still got in as target element.